manifests/m/Microsoft/WindowsSDK/10/0/22621/10.0.22621.3/Microsoft.WindowsSDK.10.0.22621.installer.yaml

```diff
@@ -0,0 +1,23 @@
+# Created with YamlCreate.ps1 v2.2.7 $debug=AUSU.CRLF.5-1-19041-2673.Win32NT
+# yaml-language-server: $schema=https://aka.ms/winget-manifest.installer.1.4.0.schema.json
+
+PackageIdentifier: Microsoft.WindowsSDK.10.0.22621
+PackageVersion: 10.0.22621.3
+MinimumOSVersion: 10.0.0.0
+InstallerType: exe
+InstallerSwitches:
+  Silent: /q
+  SilentWithProgress: /q
+UpgradeBehavior: uninstallPrevious
+AppsAndFeaturesEntries:
+- DisplayName: Windows Software Development Kit - Windows 10.0.22621.755
+  Publisher: Microsoft Corporation
+  DisplayVersion: 10.1.22621.755
+  ProductCode: '{bbdcfd1a-000e-4b9d-af4f-07049d1e9a47}'
```

Comment:
Is this necessary? As new revisions come out - if any come out - this should be different. Can this just be elided?

Reply:
> Is this necessary? As new revisions come out - if any come out - this should be different. Can this just be elided?

Unfortunately, yes, it is necessary. Because the version written to control panel is different than the actual package version, the `DisplayVersion` key is needed for WinGet to accurately interpret what the version is - without it, users could be caught in a loop where an upgrade is always available even if they are on the latest version; or (potentially) worse, not be notified when a newer version is available for install. The product code is needed to differentiate between different versions of the Windows SDK, and this prevents winget from accidentally mapping 10.0.22000 into 10.0.22621 since they both start with "Windows Software Development Kit". The DisplayName is technically not needed, but makes it easier for the cli to match an installed package version to an available package version. The Publisher could be omitted but increases the confidence heuristic similar to the DisplayName.